src/js/messaging/containers/Main.jsx

```diff
@@ -35,13 +41,66 @@ class Main extends React.Component {
     this.props.setNewFolderName(field);
   }
 
-  handleSubmit() {}
+  handleCreateNewFolderModal() {
+    // Reset the form / state object when closed
+    const resetForm = { value: '', dirty: false };
+    this.props.setNewFolderName(resetForm);
+    this.props.toggleCreateFolderModal();
+  }
+
+  handleSubmit(domEvent) {
+    domEvent.preventDefault();
+    const input = domEvent.target.getElementsByTagName('input')[0];
+    // If, by chance, the veteran has submitted this form without touching the
+    // folder name field, trigger an action that will trigger an error.
```

Comment:
I tried that and couldn't figure out a way to do it since this modal is added and removed from the DOM when visible/not. Nothing I tried work. Maybe with `componentDidUpdate`? Can revisit later.